cryptography/src/transcript.rs

```diff
@@ -184,6 +184,12 @@ impl Transcript {
     /// Like [Self::commit], except that subsequent calls to [Self::append] or [Self::commit] are
     /// considered part of the same message.
     ///
+    /// [Self::commit] needs to be called before calling any of:
+    /// - [Self::summarize],
+    /// - [Self::noise],
+    /// - [Self::fork],
+    /// in order to avoid having uncommitted data.
```

Comment:
Perhaps it's better to not mention other functions explicitly as you may add a function and forget to mention it here. So I suggest something like:
```
[Self::append] should not be the most-recently called function before calling any function besides [Self::commit], otherwise those functions may panic due to having uncommitted data.
```